### ISSUE TITLE
Allow defining .m2 path for improved caching

### ISSUE THREAD

**Issue by jeromewaibel** [paketo-buildpacks/maven#58]
* What were you attempting to do?

As stated [here ](https://stackoverflow.com/questions/37785154/how-to-enable-maven-artifact-caching-for-gitlab-ci-runner) it is very common for gitlab CI builds to redefine the .m2 path, as it must be located within the build directory to be cachable by gitlab.

Alas, your build-pack is referring to ~/.m2 only, which is empty, when trying to re-use downloaded dependencies.  This leads to re-downloading every dependency on every build.

Please provide a method for overrding the .m2 location when using this build pack.

**Comment by dmikusa-pivotal** [comment]
Can you provide some more details here? The full command you're running along with the output of two runs (one the first run where caching wouldn't happen and the second where caching should have happened) would be helpful to understand more about what is happening.

While the buildpack is referring to `~/.m2`, it's more complicated than that. There is actually a symlink between that location and the actual location, which is elsewhere (and has to be). Plus, all of these paths are only relevant in the build container environment, not outside the container. 

I don't know exactly how Gitlab is running the build, but typically the buildpack is completely isolated in a container and doesn't look at, modify or even have access to the `~/.m2` directory on the host (which I believe is what's being referred to in the link). The cached content that persists between buildpack builds is normally handled with docker volumes. Again, Gitlab may be doing this differently, if you can provide any context or doc links for Gitlab to explain how you have Gitlab configured to run your builds that would help too.

Thanks!

**Comment by jeromewaibel** [comment]
Thank you for your feedback. I think I misunderstood how the .m2 is handled. Somehow the statement "Links the ~/.m2 to a layer for caching" from the project README gave me the impression that it links the _hosts_ .m2 into the docker container via volume mount.

I also learned that the caching docker volume the buildpack is creating, is named after the target image name of your build. I made the mistake to name the target image "something:$GITSHA" so every CI build of course had a different image name, and thus the caching volumes never got reused.

I'm closing this issue as there is no reason for my request any more.
